### ISSUE TITLE
Invalid handles in HealthCheck

### ISSUE THREAD

**Issue by katjameden** [clarinsi/clarin-dspace#35]
In the latest HealthCheck for the repository, we found the following issue. 
These invalid Handle IDs indeed do not exist in the repository, but none of the items seems to be missing from the repository.
#### PID check [took: 0s] [# lines: 109]
====
Exception occurred!
java.lang.IllegalArgumentException: Illegal pid.service type
Reason: Warning: PID service type is not defined or wrong

Total count: 323
Invalid handles count: 56

Invalid handles:
----------------
Handle ID    Handle                              Res. type ID    Resource ID    URL
1027          11356/1027                          2             null          null
1098          11356/1098                          2             null          null
1131          11356/1131                          2             null          null
1134          11356/1134                          2             null          null
1082          11356/1082                          2             null          null
1106          11356/1106                          2             null          null
1147          11356/1147                          2             null          null
1148          11356/1148                          2             null          null
1107          11356/1107                          2             null          null
1108          11356/1108                          2             null          null
1100          11356/1100                          2             null          null
1111          11356/1111                          2             null          null
1117          11356/1117                          2             null          null
1153          11356/1153                          2             null          null
1163          11356/1163                          2             null          null
1164          11356/1164                          2             null          null
1162          11356/1162                          3             null          null
1161          11356/1161                          4             null          null
1168          11356/1168                          2             null          null
1132          11356/1132                          2             null          null
1146          11356/1146                          2             null          null
1118          11356/1118                          2             null          null
1093          11356/1093                          2             null          null
1099          11356/1099                          2             null          null
1101          11356/1101                          2             null          null
1102          11356/1102                          2             null          null
1103          11356/1103                          2             null          null
1104          11356/1104                          2             null          null
1113          11356/1113                          2             null          null
1298          11356/1298                          2             null          null
1299          11356/1299                          2             null          null
1297          11356/1297                          3             null          null
1296          11356/1296                          4             null          null
1173          11356/1173                          2             null          null
1184          11356/1184                          2             null          null
1126          11356/1126                          2             null          null
1191          11356/1191                          2             null          null
1301          11356/1301                          2             null          null
1302          11356/1302                          2             null          null
1303          11356/1303                          2             null          null
1304          11356/1304                          2             null          null
1305          11356/1305                          2             null          null
1306          11356/1306                          2             null          null
1307          11356/1307                          2             null          null
1212          11356/1212                          2             null          null
1313          11356/1313                          2             null          null
1216          11356/1216                          2             null          null
1314          11356/1314                          2             null          null
1315          11356/1315                          2             null          null
1316          11356/1316                          2             null          null
1229          11356/1229                          2             null          null
1234          11356/1234                          2             null          null
1283          11356/1283                          2             null          null
1203          11356/1203                          2             null          null
1343          11356/1343                          2             null          null
1344          11356/1344                          2             null          null

**Comment by cyplas** [comment]
Yes, this has actually been reported in the healthcheck for as far as I can remember.

I took a look in the database directly now, and can confirm, in line with the healthcheck, that there are 323 rows in the handle table, 56 of which do not have a resource_id. Google suggests (http://dspace.2283337.n4.nabble.com/HANDLE-table-entries-without-resource-id-td4676466.html) that "it's a feature, not a bug", i.e., normal and desired behaviour when an item gets deleted. If that's the case, then the fix here should probably be an absent or less alarming healthcheck section (title). I will ask Lindat about this.

**Comment by TomazErjavec** [comment]
> Yes, this has actually been reported in the healthcheck for as far as I can remember.

You are absolutely right, I did check a previous heathcheck before and didn't find it, that is why we reported it. But, obviously, my healtcheckcheck was not very good, hm. 

> normal and desired behaviour when an item gets deleted. 

That is the strange part: as far as I remember, I only ever deleted 2 items from the WebLicht collection (and was told off for it, as it crashed WebLicht). And I cannot possibly imagine why I would delete 56. Except if this is some relict from the very first testing phase with Jozef.

> I will ask Lindat about this.

This would be nice, a small step towards a more readable healthcheck :)

**Comment by cyplas** [comment]
> That is the strange part: as far as I remember, I only ever deleted 2 items from the WebLicht collection (and was told off for it, as it crashed WebLicht). And I cannot possibly imagine why I would delete 56. Except if this is some relict from the very first testing phase with Jozef.

Ondrej gave me a couple of potential leads, one of which I suspect might apply: Might it be that you're only remembering deleting items that were published and not merely *unfinished submissions*, which can be casually deleted (https://www.clarin.si/repository/xmlui/submissions)? Since clarin-dspace added support for providing pids even for unfinished submissions (https://github.com/ufal/clarin-dspace/releases/tag/lindat-v1.3-dspace-v5.5), if lr.reserve.pid.on.start is true (as is the case for us), such submissions immediately get pids. I tested deleting such a submission on beta, and when I tried to visit its URL afterwards, I got the same pink "Page not found" message as for the invalid handles on www.

**Comment by TomazErjavec** [comment]
Good lead - but not because I did it, but who knows what other submitters did. But let's assume this will remain shrouded in mystery. 
10^6$ question: Is there anyway to back-fix this? Even only for the two that were in fact published, but later deleted? 

Btw, somewhat related:
cf. https://curate.acdh.oeaw.ac.at/collection/Slovenian_language_resource_repository_CLARIN_SI.html
and particularly
https://curate.acdh.oeaw.ac.at/statistics/Slovenian_language_resource_repository_CLARIN_SI/404
The first one is from an inoperating WebLicht service, but the other two are some weird test.

**Comment by cyplas** [comment]
> 10^6$ question: Is there anyway to back-fix this? Even only for the two that were in fact published, but later deleted?

But given https://github.com/clarinsi/clarin-dspace/issues/35#issuecomment-689560074, what would be an appropriate fix?

> Btw, somewhat related:
> cf. https://curate.acdh.oeaw.ac.at/collection/Slovenian_language_resource_repository_CLARIN_SI.html
> and particularly
> https://curate.acdh.oeaw.ac.at/statistics/Slovenian_language_resource_repository_CLARIN_SI/404
> The first one is from an inoperating WebLicht service, but the other two are some weird test.

Hmm, can you clarify what the issue is here? I suspect that these URLs are not showing the same as they did when you reported this.

**Comment by TomazErjavec** [comment]
The second link they changed, now it is https://curate.acdh.oeaw.ac.at/statistics/Slovenian_language_resource_repository_CLARIN_SI/Broken

But, yes, now they are moaning about bitstreams, which is weird.

**Comment by cyplas** [comment]
> But, yes, now they are moaning about bitstreams, which is weird.

Maybe it just happened that something was wrong with CLARIN.SI when their calls happened to be called, as sometimes happens with our healthcheck link checker? According to question 9 of their faq (https://curate.acdh.oeaw.ac.at/faq), their database checks millions of links and it takes a couple of months to do one cycle ... I would thus suggest not to worry about this, if the number is small, especially if the links actually work and/or our own curation tasks don't find anything wrong.
